.github/workflows/report-vulnerability-in-gh.yaml

```diff
@@ -0,0 +1,78 @@
+      issue-labels:
+        description: A comma separated list of labels
+        required: false
+        type: string
+        default: "bug"
+      image-name:
+        description: "Name of the oci-image as saved in Dockerhub or in the docker cache.
+          It consists of <rock-name>:<rock-version>."
+        required: true
+        type: string
+
+jobs:
+  report-vulns:
+    name: Create or edit issues for reporting vulnerabilities
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Install tools
+        run: |
+          sudo snap install gh
+          sudo snap install jq
+
+      - name: Generate image name
+        id: image-name
+        run: |
+          IMAGE_NAME=$(echo ${{ inputs.image-name }} | sed 's/\:/-/g')
+          echo "image-name=${IMAGE_NAME}" >> "$GITHUB_OUTPUT"
+
+      - name: Get issue number if exists
+        id: get-issue-number
+        run: |
+          export GH_TOKEN=${{ secrets.GH_TOKEN }}
+          EXPECTED_TITLE=$(echo "${{ inputs.issue-title }} ${{ inputs.image-name }}")
+          ISSUE_NUMBER=$(gh issue list --repo $GITHUB_REPOSITORY --limit 500 --json "number,title" | jq -r --arg expected_title "$EXPECTED_TITLE" '.[] | select(.title == $expected_title) | .number')
+          echo "issue-number=$ISSUE_NUMBER" >> $GITHUB_OUTPUT
+
+      - name: Download report
+        uses: actions/download-artifact@v4
+        with:
+          name: trivy-report-${{ steps.image-name.outputs.image-name }}
+
+      - name: Issue body
+        id: issue-body
+        run: |
+          set -xeu
+          EXPECTED_TITLE=$(echo "${{ inputs.issue-title }} ${{ inputs.image-name }}")
+          echo "## $EXPECTED_TITLE" > issue.md
+          echo "" >> issue.md
+          echo "\`\`\`" >> issue.md
+          cat trivy-report-${{ steps.image-name.outputs.image-name }}.txt >> issue.md
+          echo "\`\`\`" >> issue.md
+          echo "" >> issue.md
+          echo -e "\nDetails: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> issue.md
+          echo "issue-body-file=issue.md" >> "$GITHUB_OUTPUT"
+
+      - name: Report failures via Github issue
+        run: |
+          export GH_TOKEN=${{ secrets.GH_TOKEN }}
+          EXPECTED_TITLE=$(echo "${{ inputs.issue-title }} ${{ inputs.image-name }}")
+          if [ -z ${{ steps.get-issue-number.outputs.issue-number }} ]; then
+            echo "---- Creating issue ----"
+            gh issue create --repo $GITHUB_REPOSITORY \
+            --title "$EXPECTED_TITLE" \
+            --label "${{ inputs.issue-labels }}" \
+            --body-file "${{ steps.issue-body.outputs.issue-body-file }}"
+          else
+            echo "---- Editing issue ${{ steps.get-issue-number.outputs.issue-number }}----"
+            gh issue edit --repo $GITHUB_REPOSITORY ${{ steps.get-issue-number.outputs.issue-number }} \
+            --title "$EXPECTED_TITLE" \
+            --body-file "${{ steps.issue-body.outputs.issue-body-file }}"
```

Comment:
In which scenario do you think the comment would be helpful?

I think that if I'm going to fix an issue, I'll just look at the report for that image and start tackling them, w/o caring too much about a changelog, but maybe you see some value, so I'm open to discuss it.

Reply:
Hmm you 're probably right, since the image that is being scanned is an immutable object. I was thinking that in the scnario where a package gets updated, the image may have less CVEs without us doing something and then it would be useful to know that this changed. Because let's say you may start working on it and then existing CVEs would have changed but you may have missed it and spend time working on sth that's already fixed.
**But** IIUC, the above scenario cannot take place since the image is immutable and the same CVEs will be present until we fix them and publish a newer one.

Reply:
> the above scenario cannot take place since the image is immutable and the same CVEs will be present until we fix them and publish a newer one.

yeh, we are testing a tag, so unless we publish a new one, the scanner will continuously scan the same image.

Originally, my idea was to just have the Github issue as `Vulnerabilities found for tensorboard-controller:1.9.0`, so `<rock name>:<rock version>` so that no matter the tag, we would always get an issue for the latest scanned image, that is, today we scan tag `1.9.0-xxxxx` and report, but tomorrow we scan `1.9.0-yyyyy` and edit the same issue. Only in that situation can a delta exist. I guess we have to think which route is best for us.

Option A - Report for every latest tag that we have.
We will get more issues and most likely they won't have a delta when the image gets rescanned. The downside of this is we'll have several issues for the same image and the chore of closing them as we release a new tag for the same image.
For example:

* Vulnerabilities found for tensorboard-controller:1.9.0-d74bfd2
* Vulnerabilities found for tensorboard-controller:1.9.0-8ujghdh

Option B (the one originally proposed in this PR) - Report per image basis
We will get one issue per image, the issue gets constantly updated as the image changes. That is, scanning `1.9.0-xxxxx` will create a new issue, then scanning `1.9.0-yyyyy` will edit the issue with a delta of CVEs. The resulting issue will just have the `Vulnerabilities found for tensorboard-controller:1.9.0` title.

Reply:
I have modified the workflow slightly to go with Option B, but let me know what you think.